packages/server/src/games.ts

```diff
@@ -135,9 +136,22 @@ export async function updateGameState(io: IOServer, roomName: string, playerActi
       addGameState(roomName, finalState);
       io.in(roomName).emit(GameEvent.CurrentState, finalState, finalPlayerAction);
       if (isMatchFinished) {
+        // TODO we need to persist the score and make it available for the next round
         io.in(roomName).emit(GameStatus.Ended, finalScore(finalState.players));
       }
       return;
     }
   }
 }
+
+export async function restartGameState(io: IOServer, roomName: string) {
+  const { players, activePlayer } = (await getRoomState(roomName)) as GameState;
+  // TODO in order to prevent users to go back to previous round while undoing
```

Comment:
Yeah, I think we need a different structure... And not shove everything into a the current room object.

Reply:
we can improve later, that's why I left some TODOs